eclipse/src/saros/Saros.java

```diff
@@ -41,9 +46,24 @@
 
   private static final String VERSION_COMPATIBILITY_PROPERTY_FILE = "version.comp"; // $NON-NLS-1$
 
+  private final IWorkbenchListener workbenchShutdownListener =
```

Comment:
I am not completely clear on the concepts in Eclipse. Is the "workbench" the loaded workspace for the current project? What is the difference between this stop notification and the method `stop(BundleContext)` below starting at line 152?

My guess would be that this is specific to the workbench and the method below is specific to the application. This would also mean that you can change the workbench without ending the application, which leads to my next question:

Is the plugin bound to the workbench? Otherwise, why do we want to tear down the whole context when the workbench is closed?

Reply:
There is only one workbench. Stop is called when the plugin should be stopped, e.g the plugin should be reloaded or is deinstalled, or the IDE shuts down.

However stop is normally called while the Display is already disposed. As we do not use Display#getDefault (which would create a new one) the result was a bunch of warnings in the log stating the xyz could not be executed.

Reply:
Ok, but do we then really need both methods dealing with the teardown of the plugin?
Could it have negative side-effects calling the teardown method `stopLifeCycle()` multiple times?

Reply:
stopLifeCycle can now be called multiple times, will only shutdown once.